core/src/main/java/com/falmeida/tasky/core/presentation/DispatcherProvider.kt

```diff
@@ -0,0 +1,10 @@
+package com.falmeida.tasky.core.presentation
+
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Inject
+
+class DispatcherProvider @Inject constructor() {
+    fun ui(): CoroutineDispatcher = Dispatchers.Main
+    fun io(): CoroutineDispatcher = Dispatchers.IO
```

Comment:
See, this `DispatcherProvider` couldn't be used for the coroutines launched in your `BaseViewModel` right now. 

Another thing: In order to really be able to swap out production dispatchers with test dispatchers, this `DispatcherProvider` would need to be an interface your classes depend on. You then have a production and test implementation of it.